Source/Krypton Toolkit/Krypton.Toolkit.Suite.Extended.DataGridView/Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj

```diff
@@ -42,21 +42,7 @@
   
   <!--TODO: Help!!-->
   <ItemGroup>
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net35'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net40'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net45'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net451'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net452'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net46'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net461'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net462'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net47'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net471'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net472'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net48'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'netcoreapp3.1'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net5'" />
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net6'" />
+    <Reference Include="System.ValueTuple" />
```

Comment:
So is this working for all TFM's, with just this line ?

Reply:
No, `<PackageReference Include="System.ValueTuple" Version="4.5.0" />` will work, **but** only for .netcoreapp3.1 and up

Reply:
If it's set to `<PackageReference Include="System.ValueTuple" Version="4.5.0" />`, you get the following build error
```
"T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj" (Restore target) (14) ->
(Restore target) ->
  T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202: Package System.ValueTuple 4
.5.0 is not compatible with net35 (.NETFramework,Version=v3.5). Package System.ValueTuple 4.5.0 supports:
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - monoandroid10 (MonoAndroi
d,Version=v1.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - monotouch10 (MonoTouch,Ve
rsion=v1.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - net461 (.NETFramework,Ver
sion=v4.6.1)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - net47 (.NETFramework,Vers
ion=v4.7)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - netcoreapp2.0 (.NETCoreAp
p,Version=v2.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - netstandard1.0 (.NETStand
ard,Version=v1.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - netstandard2.0 (.NETStand
ard,Version=v2.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - portable-net40+sl4+win8+w
p8 (.NETPortable,Version=v0.0,Profile=Profile36)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - uap10.0.16299 (UAP,Versio
n=v10.0.16299)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - xamarinios10 (Xamarin.iOS
,Version=v1.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - xamarinmac20 (Xamarin.Mac
,Version=v2.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - xamarintvos10 (Xamarin.TV
OS,Version=v1.0)
T:\Development\Krypton\Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj : error NU1202:   - xamarinwatchos10 (Xamarin
.WatchOS,Version=v1.0)
```

Reply:
Ok, I'm taking a few days away from the computer, Back Thursday
I'll look at things then, if you don't find a nice solution in the mean time.

Reply:
2 ways to solve it:
1) replace `<PackageReference Include="System.ValueTuple" />` with `<PackageReference Include="Portable.System.ValueTuple" Version="1.0.1" />`
2) remove code that is returning a tuple and use a return and a reference (Or 2 references)

Pro/Cons
1) Requires a Nuget dependency - third party dll (MIT) rather than an MS Supported dll (MS-PL)
2) Code looks ugly but removes the references

What choice would you like completed ?

Reply:
Option 3)
From the developers [GitHub](https://github.com/igor-tkachev/Portable.System.ValueTuple)
_"Also if you do not need any dependencies, then just copy the Portable.System.ValueTuple.cs file into your project. In this case it will be compiled as an internal class an will not be conflicted with other packages"_

Reply:
Or Option 4)
Re-steal the latest `System.ValueTuple` and remove any code that we do not need (i.e. anything that goes above 2 entries) and embed directly into the DataGridView project

Reply:
@Wagnerp So which option ?